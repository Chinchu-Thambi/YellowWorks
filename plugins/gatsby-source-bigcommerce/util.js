import axios from 'axios';
import {
  map,
  range,
  pluck,
  flatten,
  find,
  propEq,
  reduce,
  append,
} from 'ramda';

export const fetchAll = async ({
  url,
  pageSize,
  params,
  clientId,
  accessToken,
}) => {
  const loadPage = async ({ page }) => {
    const {
      data: {
        data,
        meta: {
          pagination: {
            current_page: currentPage,
            total_pages: totalPages,
          },
        },
      },
    } = await axios({
      url,
      params: {
        ...params,
        page,
        limit: pageSize,
      },
      headers: {
        Accept: 'application/json',
        'X-Auth-Client': clientId,
        'X-Auth-Token': accessToken,
      },
    });

    return {
      data,
      currentPage,
      totalPages,
    };
  };

  const {
    data: firstPageData,
    currentPage,
    totalPages,
  } = await loadPage({ page: 1 });

  let remainingPagesData = [];

  if (currentPage < totalPages) {
    // create an array (loadingPages) of promises (loadPage())
    // from currentPage to totalPages (exclusive)
    const loadingPages = await Promise.all(map(
      (page) => loadPage({ page: page + 1 }),
    )(range(currentPage, totalPages)));

    // await for all of the products to be loaded, pick the data out of them.
    remainingPagesData = pluck('data')(loadingPages);
  }

  return flatten([firstPageData, remainingPagesData]);
};

export const createGatsbyNode = ({
  data, type, nodeId, createNode, createContentDigest,
}) => createNode(
  {
    ...data,
    id: nodeId,
    parent: null,
    children: [],
    internal: {
      type,
      content: JSON.stringify(data),
      contentDigest: createContentDigest(data),
    },
  },
);

export const linkCategories = ({
  categories, createNodeId,
}) => ({
  bigCommerceCategories: categories,
  categories: undefined,
  categories___NODE: map(
    (categoryId) => createNodeId(`bigcommerce-category-${categoryId}`),
  )(categories),
});

const getBySKU = (sku) => find(propEq('sku', sku));

export const linkModifiers = ({
  products,
  modifiers,
  createNodeId,
}) => ({
  bigCommerceModifiers: modifiers,
  modifiers: undefined,
  modifiers___NODE: reduce(
    (acc, modifier) => {
      const modifierProduct = getBySKU(modifier.display_name.replace('sku.', ''))(products);

      if (!modifierProduct) {
        return acc;
      }

      return append(createNodeId(`bigcommerce-product-${modifierProduct.id}`))(acc);
    }, [],
  )(modifiers),
});
