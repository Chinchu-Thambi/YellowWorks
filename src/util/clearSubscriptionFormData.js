/* globals localStorage */

import { storageAvailable } from '.';

const clearSubscriptionFormData = () => {
  if (storageAvailable()) {
    Object.keys(localStorage).forEach((key) => {
      if (
        key === 'storageOrderDetails'
        || key.startsWith('storageFormData')
        || key.startsWith('formData-') // still needed because onboarding uses it, yet.
      ) {
        localStorage.removeItem(key);
      }
    });
  }
};

export default clearSubscriptionFormData;
