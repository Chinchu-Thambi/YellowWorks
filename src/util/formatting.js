const zeroBasedMonths = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];

export const monthIndexToString = (monthIndex) => zeroBasedMonths[monthIndex];

/**
 * Returns date of format `MMM 'yy` e.g. `Jun '19`
 * @param date
 */
export const formatDateToMonthYear = (date) => `${monthIndexToString(date.getMonth())} '${date.getFullYear().toString().substr(-2)}`;

/**
 * Returns date of format 'dd MMM yyyy' e.g. '2 Jun 2019'
 * @param date
 */
export const formatDateWithMonthName = (date) => `${date.getDate()} ${monthIndexToString(date.getMonth())} ${date.getFullYear().toString()}`;

const padNumber = (number) => (number < 10 ? `0${number}` : `${number}`);

/**
 * Returns `YYYYMM` e.g. August 2020 would be `2020-08`
 */
export const formatYYYYdashMM = (date) => `${date.getFullYear()}-${padNumber(date.getMonth() + 1)}`;

/**
 * Returns `YYYYMM` e.g. August 2020 would be `202008`
 */
export const formatYYYYMM = (date) => `${date.getFullYear()}${padNumber(date.getMonth() + 1)}`;

export const toSentenceCase = (str) => {
  if (!str) return str;

  const newStr = str.toLowerCase();
  const firstLetterCapitalized = newStr.substring(0, 1).toUpperCase();

  if (newStr.length > 1) {
    return firstLetterCapitalized + newStr.substring(1);
  }
  return firstLetterCapitalized;
};

export const utcDate = (year, month0based, day1based) => new Date(Date.UTC(year, month0based, day1based));
