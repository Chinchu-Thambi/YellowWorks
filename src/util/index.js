/* global window, DOMException, document */

import React from 'react';
import showdown from 'showdown';

import { importableModules } from '../services/useModule';

export {
  default as theme,
  mediaBreakpointUp,
} from './theme';
export {
  default as errorText,
  infos as infoText,
  getErrorString,
} from './texts';

const isSingleLine = (string) => (string !== undefined ? !string.includes('\n') : '');

const mdConverter = new showdown.Converter();

mdConverter.setOption('tables', true);

export const parseMd = (md) => {
  if (!md) {
    return '';
  }

  let mdString = md;

  if (typeof md === 'object') {
    [mdString] = Object.values(md);
  }

  let parsedString = mdConverter.makeHtml(mdString);

  if (isSingleLine(mdString)) {
    // remove outer p tag from single line md
    parsedString = parsedString.replace(/<\/?p>/g, '');
  }

  // eslint-disable-next-line react/no-danger
  return <span dangerouslySetInnerHTML={{ __html: parsedString }} />;
};

export const getProportionalHeight = ({
  initialWidth, initialHeight, targetWidth,
}) => (initialHeight * targetWidth) / initialWidth;

export const storageAvailable = () => {
  let storage;

  if (typeof window !== 'undefined') {
    storage = window.localStorage;
  }

  try {
    const x = '__storage_test__';
    storage.setItem(x, x);
    storage.removeItem(x);
    return true;
  } catch (e) {
    if (typeof DOMException === 'undefined') {
      return false;
    }

    return e instanceof DOMException && (
      // everything except Firefox
      e.code === 22
      // Firefox
      || e.code === 1014
      // test name field too, because code might not be present
      // everything except Firefox
      || e.name === 'QuotaExceededError'
      // Firefox
      || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
      // acknowledge QuotaExceededError only if there's something already stored
      && storage && storage.length !== 0;
  }
};

export const formatPrice = (price = 0) => (typeof price === 'number' ? `$${price.toFixed(2)}` : `$${price}`);

export const formatRecurringPrice = (price, recurring, withGst, annual) => {
  const priceFormatted = formatPrice(price);
  if (priceFormatted === '$0.00') {
    return 'Free';
  }

  if (recurring) {
    if (withGst) {
      return `${priceFormatted} / month + GST`;
    }
    if (annual) {
      return `${priceFormatted} / year`;
    }
    return `${priceFormatted} / month`;
  }
  if (withGst) {
    return `${priceFormatted} + GST`;
  }
  return priceFormatted;
};

// ex input date object
// ex out: "13/08/2019"
export const reformatDate = (dateObj) => (
  dateObj ? dateObj.toLocaleDateString('en-NZ') : ''
);

export const useLocalStorage = (key, initialValue) => {
  // from https://usehooks.com/useLocalStorage/
  // State to store our value
  // Pass initial state function to useState so logic is only executed once
  const [storedValue, setStoredValue] = React.useState(() => {
    try {
      // Get from local storage by key
      if (storageAvailable()) {
        const item = window.localStorage.getItem(key);
        // Parse stored json or if none return initialValue
        return item ? JSON.parse(item) : initialValue;
      }
      return initialValue;
    } catch (error) {
      // If error also return initialValue
      console.log(error);
      return initialValue;
    }
  });

  // Return a wrapped version of useState's setter function that ...
  // ... persists the new value to localStorage.
  const setValue = (value) => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore = value instanceof Function ? value(storedValue) : value;
      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      if (storageAvailable()) {
        window.localStorage.setItem(key, JSON.stringify(valueToStore));
      }
    } catch (error) {
      // A more advanced implementation would handle the error case
      console.log(error);
    }
  };

  return [storedValue, setValue];
};

// TODO: remove this and user version inside of auth/services
export const isStaff = async () => {
  try {
    const { Auth } = await importableModules['@aws-amplify/auth']();
    const user = await Auth.currentSession();
    return user.idToken.payload.email.endsWith('@yellow.co.nz');
  } catch (err) {
    // console.log('isStaff', err);
    return false;
  }
};


export const isTesting = () => document.location.search.includes('utm_isTesting=yes');

export const getNextWeekday = () => {
  const dt = new Date();
  dt.setDate(dt.getDate() + 1);

  if (dt.getDay() === 6) { // if Saturday
    dt.setDate(dt.getDate() + 2);
  }

  if (dt.getDay() === 0) { // if Sunday
    dt.setDate(dt.getDate() + 1);
  }

  return dt;
};

export const getNextMonth = ({
  baseNextWeekday = false,
} = {}) => {
  let dt = new Date();

  if (baseNextWeekday) {
    dt = getNextWeekday();
  }

  dt.setFullYear(
    dt.getFullYear(),
    dt.getMonth() + 1,
    1,
  );

  return dt;
};

export const getEndOfMonth = ({
  baseNextWeekday = false,
} = {}) => {
  let dt = new Date();

  if (baseNextWeekday) {
    dt = getNextWeekday();
  }

  dt.setFullYear(
    dt.getFullYear(),
    dt.getMonth() + 1,
    0,
  );

  return dt;
};

export const capitalize = (string = '') => `${string.charAt(0).toUpperCase()}${string.substr(1).toLowerCase()}`;

export const shuffle = (array) => {
  let counter = array.length;
  const shuffled = [...array];

  // While there are elements in the array
  while (counter > 0) {
    // Pick a random index
    const index = Math.floor(Math.random() * counter);

    // Decrease counter by 1
    counter -= 1;

    // And swap the last element with it
    const temp = shuffled[counter];
    shuffled[counter] = shuffled[index];
    shuffled[index] = temp;
  }

  return shuffled;
};
