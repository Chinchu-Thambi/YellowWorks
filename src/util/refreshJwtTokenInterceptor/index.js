import refreshJwtToken from './refreshJwtTokenInterceptor';

export default refreshJwtToken;
