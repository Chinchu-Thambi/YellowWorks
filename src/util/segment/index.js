/* globals window document */

import PropTypes from 'prop-types';
import { getTrackingProducts } from './trackingFormats';

export const segmentIdentify = ({ id, payload: { email, firstname, lastname } }) => {
  if (typeof window !== 'undefined') {
    const { analytics } = window;
    if (analytics) {
      analytics.identify(id, { email, firstname, lastname });
    }
  }
};

export const segmentTrack = ({
  event,
  payload,
}) => {
  if (typeof window !== 'undefined') {
    const { analytics } = window;
    if (analytics) {
      switch (event) {
        case 'Product Added': {
          analytics.track(event, {
            nonInteraction: 1,
            cartId: payload.cartId,
            cartedProducts: payload.cartedProducts.map(getTrackingProducts),
            addedProducts: payload.addedProducts.map(getTrackingProducts),
            clientId: payload.clientId,
            hubspotId: payload.hubspotId,
            isStaff: payload.isStaff,
          });
          break;
        }
        case 'Order Completed': {
          analytics.track(event, {
            nonInteraction: 1,
            jwtToken: payload.jwtToken,
            orderId: payload.orderId,
            checkoutId: payload.checkoutId,
            total: payload.total,
            revenue: payload.revenue,
            tax: payload.tax,
            currency: payload.currency,
            clientId: payload.clientId,
            hubspotId: payload.hubspotId,
            isStaff: payload.isStaff,
            label: payload.label,
            products: payload.products.map(getTrackingProducts),
            category: payload.category,
            action: payload.action,
            dimension: payload.dimension,
            cartId: payload.cartId,
          });
          break;
        }
        case 'cart.created': {
          analytics.track(event, {
            nonInteraction: 1,
            cartedProducts: payload.cartedProducts.map(getTrackingProducts),
            cartId: payload.cartId,
            jwtToken: payload.jwtToken,
            category: payload.category,
            clientId: payload.clientId,
            hubspotId: payload.hubspotId,
            isStaff: payload.isStaff,
            label: payload.label,
          });
          break;
        }
        case 'cart.updated': {
          analytics.track(event, {
            nonInteraction: 1,
            cartedProducts: payload.cartedProducts.map(getTrackingProducts),
            cartId: payload.cartId,
            jwtToken: payload.jwtToken,
            category: payload.category,
            clientId: payload.clientId,
            hubspotId: payload.hubspotId,
            isStaff: payload.isStaff,
            label: payload.label,
          });
          break;
        }
        case 'Product Removed': {
          analytics.track(event, {
            cartId: payload.cartId,
            cartedProducts: payload.cartedProducts.map(getTrackingProducts),
            nonInteraction: 1,
            clientId: payload.clientId,
            hubspotId: payload.hubspotId,
            isStaff: payload.isStaff,
            productsRemoved: payload.productsRemoved.map(getTrackingProducts),
          });
          break;
        }
        case `Search Ads Step ${payload.step} Reached`: {
          const dataObject = {
            nonInteraction: 1,
            isStaff: payload.isStaff,
            properties: {
              email: payload.email,
              firstname: payload.firstname,
              lastname: payload.lastname,
              ynz_sem_step: payload.ynz_sem_step,
              ...payload.data,
            },
          };
          analytics.track(event, dataObject);
          break;
        }
        default:
        {
          analytics.track(event);
        }
      }
    }
  }
};

export const segmentIdentifyAndTrack = ({
  event, id, payload,
}) => {
  if (typeof window !== 'undefined') {
    const { analytics } = window;
    if (analytics) {
      analytics.identify(id, { ...payload });
      analytics.track(event);
    }
  }
};

let eventListenerAdded = false;

export const trackGlobalClickEvents = () => {
  if (eventListenerAdded) {
    return;
  }

  document.addEventListener('click', (event) => {
    const {
      type: category, target: {
        innerHTML, className, nodeName, alt, href,
      },
    } = event;

    let label = innerHTML;
    if (nodeName === 'IMG' && label === '') {
      label = alt;
    }

    let action = '';

    if (nodeName === 'A' || nodeName === 'BUTTON') {
      if (className.includes('Button-')) action = 'Button Click';
      else action = 'Link Click';
      segmentTrack({
        event: action,
        payload: {
          category, label, action, dimension: href,
        },
      });
    }
  }, false);

  eventListenerAdded = true;
};

segmentIdentify.propTypes = {
  id: PropTypes.string.isRequired,
  payload: PropTypes.shape().isRequired,
};

segmentTrack.propTypes = {
  event: PropTypes.string.isRequired,
  payload: PropTypes.shape().isRequired,
};

segmentIdentifyAndTrack.propTypes = {
  event: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  payload: PropTypes.shape().isRequired,
};

export default segmentTrack;
