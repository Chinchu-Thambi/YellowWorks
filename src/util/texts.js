const errors = {
  cardHolderMissing: 'Please enter name on card.',
  requiredFieldMissing: 'Please enter all required fields.',
  termsNotAccepted: 'Please accept our Terms of Business and Additional Terms.',
  cardAuthorizationNotGiven: 'Please accept our Credit/Debit Card Registration Terms.',
  defaultError: 'Oops. We didn\'t anticipate this taking so long.',
};

export const infos = {
  submit: 'Order Submitting...',
};

export const getErrorString = (error) => {
  let msg;
  if (error instanceof Array && error.length > 0) {
    const { message } = error[0];
    msg = message;
  } else if (error instanceof Object) {
    const { message } = error;
    msg = message;
  } else {
    msg = error;
  }

  const errorMapping = {
  };
  const friendlyError = errorMapping[msg] ? errorMapping[msg] : errors.defaultError;
  return friendlyError;
};

export default errors;
