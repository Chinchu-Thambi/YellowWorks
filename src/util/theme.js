/* eslint-disable prefer-destructuring */
/* globals window */
import React from 'react';
import * as t from '../services/theme.json';

/** **********************************************************************************************
 * @deprecated
 *
 * For use with older rebass and styled components only.
 *
 * Use tailwind for styling new components. Refer to tailwind.config.js and services/theme.json.
 ************************************************************************************************ */


/**
 * SPEC REFERENCE AT https://github.com/system-ui/theme-specification
 */

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
const containerWidth = [
  1, // 100%
  '33.75rem', // 540px
  '43.75rem', // 720px
  '58.75rem', // 940px
  '73.75rem', // 1140px
  '90.00rem', // 1440px
];

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
const colors = {
  blue: '#07c', // palette.accent[4][0]
  brand: '#1F2A44', // palette.contrast[0]
  brandLight: '#343E55', // palette.contrast[1]
  primary: '#FBE122', // palette.brand[0]
  cerulean: '#00aded', // palette.accent[3][0]
  azure: '#345DA4', // palette.accent[4][0]
  black: '#000', // palette.contrast[0]
  dolphin: '#667', // palette.contrast[1]
  raven: '#797F8F', // palette.contrast[2]
  dustyGray: '#999', // palette.contrast[3]
  spunPearl: '#aab', // palette.contrast[3]
  silver: '#ccc', // palette.contrast[4]
  mischka: '#ddd', // palette.contrast[4]
  concrete: '#f2f2f2', // palette.contrast[5]
  pleinDeVie: '#BABABA', // palette.contrast[3]
  athensGray: '#f3f5f8', // palette.contrast[5]
  gallery: '#eee', // palette.base[3]
  lightGray: '#f6f6ff', // palette.contrast[5]
  white: '#fff', // palette.base[0],
  yellow: '#F9E01A',
};

const palette = {
  base: [
    t.colors.base['100'],
    t.colors.base['200'],
    t.colors.base['300'],
    t.colors.base['400'],
  ],
  brand: [ // The Optimist
    t.colors.brand['600'],
    t.colors.brand['500'],
    t.colors.brand['400'],
    t.colors.brand['300'],
    t.colors.brand['200'],
    t.colors.brand['100'],
  ],
  contrast: [ // Midnight
    t.colors.contrast['600'],
    t.colors.contrast['500'],
    t.colors.contrast['400'],
    t.colors.contrast['300'],
    t.colors.contrast['200'],
    t.colors.contrast['100'],
  ],
  accent: [
    [ // Crimson
      t.colors.crimson['600'],
      t.colors.crimson['500'],
      t.colors.crimson['400'],
      t.colors.crimson['300'],
      t.colors.crimson['200'],
      t.colors.crimson['100'],
    ],
    [ // Fern
      t.colors.fern['600'],
      t.colors.fern['500'],
      t.colors.fern['400'],
      t.colors.fern['300'],
      t.colors.fern['200'],
      t.colors.fern['100'],
    ],
    [ // Uplift
      t.colors.uplift['600'],
      t.colors.uplift['500'],
      t.colors.uplift['400'],
      t.colors.uplift['300'],
      t.colors.uplift['200'],
      t.colors.uplift['100'],
    ],
    [ // Pacific
      t.colors.pacific['600'],
      t.colors.pacific['500'],
      t.colors.pacific['400'],
      t.colors.pacific['300'],
      t.colors.pacific['200'],
      t.colors.pacific['100'],
    ],
    [ // Magnificent
      t.colors.magnificent['600'],
      t.colors.magnificent['500'],
      t.colors.magnificent['400'],
      t.colors.magnificent['300'],
      t.colors.magnificent['200'],
      t.colors.magnificent['100'],
    ],
    [ // Violet
      t.colors.violet['600'],
      t.colors.violet['500'],
      t.colors.violet['400'],
      t.colors.violet['300'],
      t.colors.violet['200'],
      t.colors.violet['100'],
    ],
  ],
};

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
// color aliases for palette.accent
palette.error = palette.accent[0];
palette.success = palette.accent[1];
palette.warn = palette.accent[2];
palette.navigation = palette.accent[3];
palette.support = palette.accent[4];

const theme = {
  breakpoints: [
    t.screens.sm,
    t.screens.md,
    t.screens.lg,
    t.screens.xl,
    t.screens['2xl'],
  ],
  containerWidth,
  fontSizes: [
    t.fontSize.sm[0],
    t.fontSize.xs[0],
    t.fontSize.base[0],
    t.fontSize['2xl'][0],
    t.fontSize['3xl'][0],
    t.fontSize['4xl'][0],
    t.fontSize['6xl'][0],
  ],
  // TODO replace usage of this...
  fontWeight: [
    t.fontWeight.normal,
    t.fontWeight.bold,
  ],
  // TODO ...with this (e.g. fontWeight="bold")
  fontWeights: { // PROPER SCHEMA
    body: t.fontWeight.normal,
    bold: t.fontWeight.bold,
  },
  zIndex: {
    dropdown: 1000,
    sticky: 1020,
    fixed: 1030,
    modalBackdrop: 1040,
    modal: 1050,
    popOver: 1060,
    tooltip: 1070,
  },
  colors,
  palette,
  space: [
    t.spacing['0'],
    t.spacing['1'],
    t.spacing['2'],
    t.spacing['3'],
    t.spacing['4'],
    t.spacing['5'],
    t.spacing['6'],
    t.spacing['7'],
    t.spacing['8'],
    t.spacing['9'],
    t.spacing['10'],
    t.spacing['11'],
    t.spacing['12'],
    t.spacing['13'],
    t.spacing['14'],
  ],
  fonts: {
    sans: t.fontFamily.sans[0],
    title: t.fontFamily.title[0],
  },
  shadows: {
    subtle: t.boxShadow.xs,
    small: t.boxShadow.sm,
    medium: t.boxShadow.md,
    large: t.boxShadow.lg,
  },
  // TODO should not have these buttons in the theme - delete later
  buttons: {
    primary: {
      color: 'black',
      backgroundColor: colors.primary,
      textTransform: 'uppercase',
      borderRadius: '40px',
      fontFamily: 'Montserrat',
      fontWeight: 500,
      fontSize: '0.875rem',
    },
    outline: {
      color: colors.blue,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
  radii: [
    t.borderRadius.none,
    t.borderRadius.default,
    t.borderRadius.md,
    t.borderRadius.lg,
  ],
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
const widthToBreakpoint = (width) => {
  if (width <= 576) return 'xs';
  if (width <= 768) return 'sm';
  if (width <= 992) return 'md';
  if (width <= 1200) return 'lg';
  if (width <= 1440) return 'xl';
  return 'xxl';
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const useBreakpoint = () => {
  if (typeof window === 'undefined') { return 'xs'; }
  // SSR for static pages will fail due to window useage - shift towards default window instead.
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [breakpoint, setBreakpoint] = React.useState(widthToBreakpoint(window.innerWidth));
  // eslint-disable-next-line react-hooks/rules-of-hooks
  React.useEffect(() => {
    const onResize = () => {
      setBreakpoint(widthToBreakpoint(window.innerWidth));
    };
    const resizeListener = window.addEventListener('resize', onResize);
    return () => {
      window.removeEventListener('resize', resizeListener);
    };
  });
  return breakpoint;
};

export default theme;

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const mediaBreakpoints = {
  xs: {
    minScreenWidth: 0,
    maxScreenWidth: 576 - 0.02,
  },
  sm: {
    minScreenWidth: theme.breakpoints[0],
    maxContainerWidth: containerWidth[1],
  },
  md: {
    minScreenWidth: theme.breakpoints[1],
    maxContainerWidth: containerWidth[2],
  },
  lg: {
    minScreenWidth: theme.breakpoints[2],
    maxContainerWidth: containerWidth[3],
  },
  xl: {
    minScreenWidth: theme.breakpoints[3],
    maxContainerWidth: containerWidth[4],
  },
  xxl: {
    minScreenWidth: theme.breakpoints[4],
    maxContainerWidth: containerWidth[5],
  },
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const mediaBreakpointUp = (size) => (
  mediaBreakpoints[size]
    ? `@media (min-width: ${mediaBreakpoints[size].minScreenWidth})`
    : ''
);
