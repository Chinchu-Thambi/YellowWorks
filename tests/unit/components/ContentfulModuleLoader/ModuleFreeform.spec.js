/* globals describe, it, expect, window, document, jest */

import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../../src/util';
import Loader from '../../../../src/components/ContentfulModuleLoader/ModuleFreeform';

describe('freeform modules', () => {
  it('renders HubspotForm', () => {
    const content = {
      portalId: '123',
      formId: '123',
    };

    const data = {
      internal: {
        content: JSON.stringify(content),
      },
    };

    const documentSpy = jest.spyOn(document, 'createElement');

    let wrapper = mount(
      <ThemeProvider theme={theme}>
        <Loader
          theme={theme}
          data={data}
          source="HubspotForm"
        />
      </ThemeProvider>,
    );

    expect(documentSpy).toHaveBeenCalledWith('script');

    window.hbspt = {
      forms: {
        create: jest.fn(),
      },
    };

    wrapper.unmount();

    wrapper = mount(
      <ThemeProvider theme={theme}>
        <Loader
          data={data}
          source="HubspotForm"
        />
      </ThemeProvider>,
    );

    expect(wrapper.find('section#hubspot-form-container-123')).toHaveLength(1);

    expect(window.hbspt.forms.create).toHaveBeenCalledWith({
      portalId: '123',
      formId: '123',
      target: '#hubspot-form-container-123',
    });

    wrapper.unmount();
  });

  it('renders GoogleParner', () => {
    const content = {
    };

    const data = {
      internal: {
        content: JSON.stringify(content),
      },
    };

    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Loader
          data={data}
          source="GooglePartner"
        />
      </ThemeProvider>,
    );

    expect(wrapper.html()).toMatchSnapshot();

    wrapper.unmount();
  });
});
