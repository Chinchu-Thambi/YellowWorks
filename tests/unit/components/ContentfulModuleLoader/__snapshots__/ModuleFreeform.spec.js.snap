// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`freeform modules renders GoogleParner 1`] = `"<div class=\\"sc-pNWxx kxXsqB\\"><div class=\\"sc-jrsJCI kptHWJ\\"><section width=\\"1,33.75rem,43.75rem,58.75rem,73.75rem,90.00rem\\" class=\\"sc-bdnylx sc-dlnjPT sc-iCoHVE llWFtj dGQNBV uryn\\"><h2>We're pretty good at this</h2><div class=\\"sc-fujyUd gblnJc\\"><div class=\\"g-partnersbadge\\" data-agency-id=\\"2298579190\\"></div><p>We are an award winning Google Premier Partner and our team of NZ-based analysts are the best at growing business online in Australasia. Let us take care of your search marketing and get the results you want!</p></div></section></div></div>"`;
